Made smooth orientation and position changes.

Reality-Dev opened the pull request:
Instead of snapping to a new orientation or anchor, now these changes are smooth.
A video of the differences can be seen here:
https://www.youtube.com/watch?v=4935SvOuZ8s&feature=youtu.be

Commit:
Rotations are smooth to and from vertical, horizontal orientations.
Animates position movement to billboard position instead of just setting new anchor.

Commit:
Removed odd scaling effect.

Commit:
Fixed vertical alignment.

Commit:
Updated ReadMe file.

maxxfrazer commented:
The video looks like a huge improvement, thanks for adding! I have a couple of suggestions, will add them now and let me know what you think when you get chance

maxxfrazer commented on Sources/FocusEntity/FocusEntity+Alignment.swift:
```diff
@@ -92,28 +93,32 @@ extension FocusEntity {
-      tempNode.simdOrientation = raycastResult.worldTransform.orientation
-      shouldAnimateAlignmentChange = true
+    
+    var targetAlignment = simd_quatf()
```
No need to initialise a simd_quatf here as a new one will be assigned in the if or else, just declare the type:
`var tagetAlignment: simd_quatf`

Reality-Dev replied:
Good catch! I made the change.

maxxfrazer commented on Sources/FocusEntity/FocusEntity+Alignment.swift:
```diff
@@ -160,9 +165,20 @@ extension FocusEntity {
+    ///Uses interpolation between orientations to create a smooth `easeOut` orientation adjustment animation.
+      internal func performAlignmentAnimation(to newOrientation: simd_quatf) {
+        //interpolate between current and target orientations
+        orientation = simd_slerp(orientation, newOrientation, 0.15)
+        let testVector : simd_float3 = [0.7,0.7,0.7]
+        let point1 = orientation.act(testVector)
+        let point2 = newOrientation.act(testVector)
+        let distanceBetweenVectors = simd_distance(point1, point2)
+        //Stop interpolating when the rotations are close enough to each other.
+        if distanceBetweenVectors < 0.03 {
+            //Stop calling this function for horiztonal and vertical surfaces.
+            isChangingAlignment = false
+        }
```
Really like how this animates, but not a big fan of the 0.7 vector, [0, 0, 1] is simple enough for the needs and is normalised. We could instead use the dot product of the normalised vectors, this way we really know how far apart the vectors are, rather than arbitrary 0.03:
```swift
orientation = simd_slerp(orientation, newOrientation, 0.15)
let testVector: simd_float3 = [0, 0, 1]
let point1 = orientation.act(testVector)
let point2 = newOrientation.act(testVector)
let vectorsDot = simd_dot(point1, point2)
// The angle between the two points is acos(vectorsDot)
// but performing acos is expensive and not necessary
// if the vectors are equal, vectorsDot = 1
// not sure on the exact angle here that works best, but:
// if they are 3º apart, .pi / 60 radians: cos(.pi / 60) = 0.99863
self.isChangingAlignment = vectorsDot < 0.999 // acos(0.999) ~= 0.256º
```
No problem if you'd rather stick with a distance between the two points, but if you agree this would be good, then remove most of the comments I've added in the code snippet except maybe the last one

Reality-Dev replied:
Hey Max! Great suggestion. I'm glad you found a way to compute this without calling acos(). The reason I do not want to use a vector along an axis is because one type of rotation (yaw, pitch, roll) will not affect the resulting vector. i.e. [0,0,1] is along the Z-Axis, and therefore roll-rotations will not change the vector resulting from .act().
For example, one of these quaternions represents a rotation of 180°  about the Z-Axis, and the other approximately 90° about the Z-Axis, but the dot product of the vectors resulting from .act() is 1.0.
```
let orientation = simd_quatf(angle: .pi, axis: [0,0,1])
let newOrientation = simd_quatf(angle: 1.57, axis: [0,0,1])

let testVector: simd_float3 = [0, 0, 1]
let point1 = orientation.act(testVector)
let point2 = newOrientation.act(testVector)

print(simd_dot(point1, point2))
//Prints 1.0
```

Maybe we could combine our solutions and use a testVector that is equivalent on all three components, and use simd_dot() instead of acos().
A value of (1 / sqrtf(3))  on all 3 components will leave it normalized.

maxxfrazer replied:
oh of course, didn't consider the rotation about the z axis. I was thinking of changing to the Y axis instead, but that wouldn't be good enough either as it rotates about that when moving from two vertical walls.
The most important thing is using a normalised vector so that the dot product still gives cos of the angle, so using this as the test vector would works too while allowing rotations in x y and z:
`let testVector = simd_normalize(simd_float3(1,1,1))`
or [0.577, 0.577, 0.577], but I don't think that level of optimising is necessary

maxxfrazer commented on FocusEntity-Example/FocusEntity-Example.xcodeproj/project.pbxproj:
```diff
@@ -327,14 +327,14 @@
 				CODE_SIGN_STYLE = Automatic;
 				DEVELOPMENT_ASSET_PATHS = "\"FocusEntity-Example/Preview Content\"";
-				DEVELOPMENT_TEAM = 278494H572;
+				DEVELOPMENT_TEAM = 9792AFNZU9;
```
Could you add a commit with this and PRODUCT_BUNDLE_IDENTIFIER empty? I try to do that with other repos to simplify the example projects

Reality-Dev replied:
Yes, thank you for pointing that out.

maxxfrazer reviewed (changes requested):
Thank you for this awesome enhancement!!

Couple of suggestions, see what you think of them + the SwiftLint suggestions please.

Reality-Dev commented:
> The video looks like a huge improvement, thanks for adding! I have a couple of suggestions, will add them now and let me know what you think when you get chance

Hey Max! Thanks for the revisions. Hopefully I will be able to fix it tomorrow.

maxxfrazer commented on Sources/FocusEntity/FocusEntity.swift:
```diff
@@ -155,6 +154,9 @@ open class FocusEntity: Entity, HasAnchoring, HasFocusEntity {
   /// Indicates if the square is currently changing its alignment.
   internal var isChangingAlignment = false
```
One last thing, let's set this to `public internal(set)`, as it's actually used now so important if someone is using it to place objects at the focus entity position (with orientation)

Reality-Dev replied:
Good point, making the change.

Commit:
Removed acos in favor of simd_dot().
Just declared the type of targetAlignment instead of unnecessarily initializing it.
Added placeholders on ReadMe to make customization process more explicit.

Reality-Dev commented:
Updated commit has been pushed. Please let me know what you think when you get chance :)

maxxfrazer commented:
LGTM, I'll update the SwiftLint changes after merging to make it easier
